hydra-node/src/Hydra/Chain.hs

```diff
@@ -54,7 +61,10 @@ instance Arbitrary HeadParameters where
   arbitrary = dedupParties <$> genericArbitrary
    where
     dedupParties HeadParameters{contestationPeriod, parties} =
-      HeadParameters{contestationPeriod, parties = nub parties}
+      HeadParameters
+        { contestationPeriod
+        , parties = sort . take maximumNumberOfParties $ nub parties
```

Comment:
Is this actually needed? The `Arbitrary Environment` is not honoring this.

Reply:
i think we should we add it to the `Arbitrary Environment` as well, don't you?

Reply:
All tests pass with this change moved to `Environment` instance for `Arbitrary` so I'd say let's move and keep this bit of logic in there.